File: helloworld.py
from pyspark import SparkConf, SparkContext

# Launch standalone project
conf = SparkConf().setAppName("HelloWorld")
sc = SparkContext(conf = conf)
lines = sc.textFile("/opt/spark/README.md")
pythonLines = lines.filter(lambda line: "Python" in line)
result = pythonLines.collect()
print("================")
for line in result:
    print(line)
print("================")

File: src/main/java/spark/HelloWorld.java
package spark;

import org.apache.log4j.Logger;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

import java.util.List;

/**
 * Created by ogre0403 on 2016/1/23.
 */
public class HelloWorld {
    private static Logger logger = Logger.getLogger(HelloWorld.class);
    public static void main(String[] args) {
        String inputFile = args[0];
        SparkConf conf = new SparkConf().setAppName("HelloWorld").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> lines = sc.textFile(inputFile);

        JavaRDD<String> pythonLines = lines.filter(
                new Function<String, Boolean>() {
                    @Override
                    public Boolean call(String line) throws Exception {
                        return line.contains("Python");
                    }
                }
        );

        List<String> result = pythonLines.collect();

        for(String s: result)
            logger.info(s);
    }
}
